test/fixtures/rules/std_rule_cases/L003.yml

```diff
@@ -877,3 +877,89 @@ test_fail_snowflake_merge_statement:
   configs:
     core:
       dialect: snowflake
+
+test_pass_hanging_indents_allowed_in_default:
+  pass_str: |
+    SELECT
+        a.line + (a.with
+                  + a.hanging_indent) as actually_ok,
+    FROM tbl as a
+  configs:
+    core:
+      dialect: ansi
+
+test_fail_disallow_hanging_indents_convert_hanging_to_normal_indent:
+  fail_str: |
+    SELECT
+        a.line + (a.with
+                  + a.hanging_indent) as actually_not_ok,
+    FROM tbl as a
+  fix_str: |
+    SELECT
+        a.line + (a.with
+            + a.hanging_indent) as actually_not_ok,
+    FROM tbl as a
+  configs:
+    core:
+      dialect: ansi
+    rules:
+      L003:
+        disallow_hanging_indents: True
+
+test_fail_disallow_hanging_indents_complex_example:
```

Comment:
Not a fan of this "complex example style" better to pick specific examples and make each one clear. Otherwise you might capture and seal unintended behaviour. In the future when people look back they wont understand the tests intention and why things are failing. I've had a few cases where the tests ended up sealing in something incorrect.